Ctl 010 merge database on contact detail

heupotato opened the pull request.

heupotato committed:
Create 'delete-contact' confirmation box

heupotato committed:
Remove moor packages and install hive packages

heupotato committed:
Add Contact database and its adapter

heupotato committed:
Remove unused import

heupotato committed:
Add Hive init() and registerAdapter to main

These functions are to init database

heupotato committed:
Open Hive Box in main to retrieve database later

heupotato committed:
Remove delete_contact_box

heupotato committed:
Modify Gender's 3rd option

heupotato committed:
Merge database to contact_detail_screen

Pass the index of contact from contact_list_screen to contact_detail_screen. Here, get database of a contact by index and then pass its data to others widgets

heupotato committed:
Update workspace.xml

lngyeen commented on lib/ui/widgets/contact_detail.dart:
```diff
@@ -1,5 +1,5 @@
-import 'package:flutter_contact_list/mockdata/mock_contact.dart';
+import 'package:flutter_contact_list/data/contact_data.dart';
 
 class Detail extends StatelessWidget {
```
class name should be 'ContactDetail'

lngyeen commented on lib/ui/screens/contact_detail_screen.dart:
```diff
@@ -1,20 +1,25 @@
 import 'package:flutter_contact_list/ui/widgets/contact_header.dart';
+import 'package:hive/hive.dart';
 
 class ContactDetail extends StatelessWidget {
```
Class name should be 'ContactDetailScreen'

lngyeen commented on lib/main.dart:
```diff
@@ -1,14 +1,36 @@
         theme: ThemeData(primaryColor: Colors.teal[700]),
-        home: ContactListScreen(),
+        home: FutureBuilder(
+          future: Hive.openBox("contacts"),
```
Should create a dedicated class to handle all database tasks. 
Put it in '/lib/storage/repositories/contacts_repository.dart'
For example:

```
class ContactsRepository {
static Future<List<Contact>> getAllContacts() {}
static Future<bool>addContact(Contact contact) {}
static Future<bool>deleteContact(Contact contact) {}
static Future<bool>updateContact(Contact contact) {}
}
```

heupotato committed:
Insert Scrollable view to avoid UI error

heupotato committed:
Fix overflow UI error

heupotato committed:
Remove toString overriden function

heupotato committed:
Add contact_repositories class to handle all database task

heupotato committed:
Update others file due to change of database class

heupotato committed:
Update contact_detail.dart

heupotato committed:
Update workspace.xml

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -15,25 +16,26 @@ class ContactListScreen extends StatefulWidget {
 
 class _ContactListScreenState extends State<ContactListScreen> {
 
+  final contactList = [];
```
Variable should always have type:
`List<Contact> contactList = [];`

lngyeen commented on lib/storage/repositories/contacts_repositories.dart:
```diff
@@ -0,0 +1,23 @@
+import 'package:hive/hive.dart';
+
+class ContactsRepository{
+  static List<Contact> getAllContacts() {
```
```suggestion
  static List<Contact> getAllContacts() {
    return Hive.box<Contact>("contacts").values.toList();
  }
```

heupotato committed:
Update contacts_repositories.dart

- Change Future<void> openbox() to Future<Box<Contact>> loadBox()
- Change all of the Box<dynamic> to Box<Contact>
- Shorten gellAllContacts function

heupotato committed:
Update main.dart

Remove FutureBuider, call loadBox function before running the app

heupotato committed:
Update contact_detail.dart

Change the class' name

heupotato committed:
Update contact_list_screen.dart

Add nullbox in case the contactList is empty

heupotato committed:
Update contact_detail_screen.dart

- Change the class' name
- Change the variable's type from int to Contact in order to pass the whole  Contact object from ContactListScreen

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -46,14 +49,15 @@ class _ContactListScreenState extends State<ContactListScreen> {
-    //         )
-    // );
+  _navigateDetail(int index){
+    if (ContactsRepository.getContactInfo(index) != null)
```
final Contact? contact = ContactsRepository.getContactInfo(index);
```suggestion
    final Contact? contact = ContactsRepository.getContactInfo(index);
if (contact != null) {
Navigator.push(
          context,
          MaterialPageRoute(
              builder: (context) => 
                  ContactDetailScreen(contact: contact)
              );
}

```

lngyeen commented on lib/storage/repositories/contacts_repositories.dart:
```diff
@@ -0,0 +1,21 @@
+
+  static List<Contact>? getAllContacts() {
+    final contactsData = getBox();
+    return contactsData.values.toList();
```
Better if use one line:

`static List<Contact>? getAllContacts() => getBox().values.toList();`

lngyeen commented on lib/storage/repositories/contacts_repositories.dart:
```diff
@@ -0,0 +1,21 @@
+  }
+  static Contact? getContactInfo(int index){
+    Box contacts = getBox();
+    return contacts.getAt(index);
```
Better if use only one line:

`static Contact? getContactInfo(int index) => getBox().getAt(index);`

lngyeen commented on lib/ui/screens/contact_list_screen.dart:
```diff
@@ -15,25 +17,26 @@ class ContactListScreen extends StatefulWidget {
-  Box _allContact() {
-    return Hive.box("contacts");
+  _getAllContacts() {
+    contactList.addAll(ContactsRepository.getAllContacts()!.toList());
```
Remove this and get contacts directly like above comment.

heupotato committed:
Update ContactRepositories

Shorten the functions

heupotato committed:
Remove unused imports

heupotato committed:
Update contact_list_screen.dart
